pkg/core/bootstrap/autoconfig.go

```diff
@@ -3,36 +3,75 @@ package bootstrap
 import (
 	"fmt"
 	"io/ioutil"
-	"net/url"
 	"os"
-	"strings"
 
 	"github.com/pkg/errors"
 
 	"github.com/kumahq/kuma/pkg/config/api-server/catalog"
 
 	kuma_cp "github.com/kumahq/kuma/pkg/config/app/kuma-cp"
-	config_core "github.com/kumahq/kuma/pkg/config/core"
 	"github.com/kumahq/kuma/pkg/core"
 	"github.com/kumahq/kuma/pkg/tls"
 )
 
 var autoconfigureLog = core.Log.WithName("bootstrap").WithName("auto-configure")
 
 func autoconfigure(cfg *kuma_cp.Config) error {
+	if err := autoconfigureTLS(cfg); err != nil {
+		return err
+	}
+	autoconfigureServersTLS(cfg)
 	autoconfigureAdminServer(cfg)
 	autoconfigureCatalog(cfg)
 	autoconfigBootstrapXdsParams(cfg)
-	if err := autoconfigureKds(cfg); err != nil {
-		return err
+	return nil
+}
+
+func autoconfigureServersTLS(cfg *kuma_cp.Config) {
+	//if cfg.XdsServer.TlsCertFile == "" {
+	//	cfg.XdsServer.TlsCertFile = cfg.General.TlsCertFile
+	//	cfg.XdsServer.TlsKeyFile = cfg.General.TlsKeyFile
+	//}
+	// todo restore in the next PR
+	if cfg.BootstrapServer.TlsCertFile == "" {
+		cfg.BootstrapServer.TlsCertFile = cfg.General.TlsCertFile
+		cfg.BootstrapServer.TlsKeyFile = cfg.General.TlsKeyFile
+	}
+	if cfg.SdsServer.TlsCertFile == "" {
+		cfg.SdsServer.TlsCertFile = cfg.General.TlsCertFile
+		cfg.SdsServer.TlsKeyFile = cfg.General.TlsKeyFile
+	}
+	if cfg.Multicluster.Global.KDS.TlsCertFile == "" {
+		cfg.Multicluster.Global.KDS.TlsCertFile = cfg.General.TlsCertFile
+		cfg.Multicluster.Global.KDS.TlsKeyFile = cfg.General.TlsKeyFile
```

Comment:
Hi @jakubdyszkiewicz , 

For cases when Kuma-cp is hosted on a VM(all behind same IP) would work but 
in case kuma-cp is hosted behind a gateway which exposes services on separate fqdn may break with these changes. 

can you kindly confirm that we can still continue to use separate certs for different services Kuma-cp exposes.

Reply:
Hey, that's why there is this `if cfg.Multicluster.Global.KDS.TlsCertFile == ""`. You can still provide your own certs.

Reply:
right.. i am little concern as autoconfigureSds is removed and in case "cfg.ApiServer.Catalog.Sds.Url" is configured differently it may not work.. 
however i agree that we can provide our certs and "cfg.ApiServer.Catalog.Sds.Url" can match the cert CN name.